File: examples/governance_static_go/domain/resolve_incident.go
package domain

import (
	"time"
	"fmt"
)

type ResolveIncident struct {
	IncidentId string `json:"incident_id"`
	Resolution string `json:"resolution"`
	RootCause string `json:"root_cause"`
}

func (c ResolveIncident) CommandName() string { return "ResolveIncident" }

func (c ResolveIncident) Execute(repo IncidentRepository) (*Incident, *ResolvedIncident, error) {
	existing, err := repo.Find(c.IncidentId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("Incident not found: %s", c.IncidentId)
	}
	existing.Resolution = c.Resolution
	existing.RootCause = c.RootCause
	if existing.Status != "investigating" && existing.Status != "mitigating" {
		return nil, nil, fmt.Errorf("cannot ResolveIncident: Incident is in %s state", existing.Status)
	}
	existing.Status = "resolved"
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := ResolvedIncident{
		AggregateID: existing.ID,
		IncidentId: c.IncidentId,
		Resolution: c.Resolution,
		RootCause: c.RootCause,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/operations_domain/lib/operations_domain/incident/commands/resolve_incident.rb
module OperationsDomain
  class Incident
    module Commands
      class ResolveIncident
        include Hecks::Command
        emits "ResolvedIncident"

        attr_reader :incident_id
        attr_reader :resolution
        attr_reader :root_cause

        def initialize(
          incident_id: nil,
          resolution: nil,
          root_cause: nil
        )
          @incident_id = incident_id
          @resolution = resolution
          @root_cause = root_cause
        end

        def call
          existing = repository.find(incident_id)
          if existing
            unless ["investigating", "mitigating"].include?(existing.status)
              raise OperationsDomain::Error, "Cannot ResolveIncident: status must be one of investigating, mitigating, got '#{existing.status}'"
            end
            Incident.new(
              id: existing.id,
              model_id: existing.model_id,
              severity: existing.severity,
              category: existing.category,
              description: existing.description,
              reported_by_id: existing.reported_by_id,
              reported_at: existing.reported_at,
              resolution: resolution,
              root_cause: root_cause,
              resolved_at: Time.now.to_s,
              status: "resolved"
            )
          else
            raise OperationsDomain::Error, "Incident not found: #{incident_id}"
          end
        end
      end
    end
  end
end
